sqsworkers/crew.py

```diff
@@ -58,6 +62,10 @@ def __init__(self, **kwargs):
         self.statsd = kwargs['statsd'] if 'statsd' in kwargs else DummyStatsd(self.logger)
         self.sentry = kwargs['sentry'] if 'sentry' in kwargs else None
         self.worker_limit = kwargs['worker_limit'] if 'worker_limit' in kwargs else 10
+        self.max_number_of_messages = kwargs['max_number_of_messages'] if 'max_number_of_messages' in kwargs else 1
```

Comment:
I haven't played with SQS in a while, but my understanding was that:

- max wait time is the maximum time it will wait for a message before returning no message
- max number of messages is how many messages it can deliver at once (but will not wait for that number to appear)

If this is true, you already have the maximum long poll, and you've actually reduced the default, changing to shorter polling - previously, it would wait 20 seconds for a message, and now it will only wait at most 5. In both cases, as soon as one new message appears, it will be delivered and the request will return, so the max_number_of_messages only has an effect if you have a large volume of messages.

If you want to decrease the requests to SQS, however, the best thing I'm assuming is to decrease threads and/or have an interval between your polls (along with increasing max_number_of_messages).

Reply:
As per my understanding, its either/or condition whichever happens first. So, if max_number_messsage=1 and wait_time=5 (default values), it will pick up a message as soon as it arrives since the first condition is satisfied. However, if we give max_number_messages=10 and wait_time=20, in this case, let us say we received 5 messages but wait time has already reached 20 seconds, in that case it will still start picking up the messages and not wait for max_number_messages to reach 10. 

@bladym  - correct me if I have misunderstood something here.

Reply:
Hi @riddhi77 , I tested this now, and my receive message calls return as soon as there are any messages, and do NOT wait until there are max number of messages. i.e. the rules are:

- if there are no messages, wait wait_time until there are messages
- if there are messages, return at at most max_number_messages

Reply:
This is easy to test via the AWS CLI with something like:

```
while true; do
  aws sqs receive-message --queue-url https://... --wait-time-seconds 20 --max-number-of-messages 10
done
```

and then

```
aws sqs send-message --queue-url https://... --message-body foo
```